File: backend/mdfe/src/model/mdfe_informacao_nfe_model.py
from src import db
from src.model.mdfe_municipio_descarrega_model import MdfeMunicipioDescarregaModel


class MdfeInformacaoNfeModel(db.Model):
    __tablename__ = 'mdfe_informacao_nfe'

    id = db.Column(db.Integer, primary_key=True)
    chave_nfe = db.Column(db.String(44))
    segundo_codigo_barra = db.Column(db.String(36))
    indicador_reentrega = db.Column(db.Integer)
    id_mdfe_municipio_descarrega = db.Column(db.Integer, db.ForeignKey('mdfe_municipio_descarrega.id'))

    mdfe_municipio_descarrega_model = db.relationship('MdfeMunicipioDescarregaModel', foreign_keys=[id_mdfe_municipio_descarrega])

    def mapping(self, data):
        self.id = data.get('id')
        self.id_mdfe_municipio_descarrega = data.get('idMdfeMunicipioDescarrega')
        self.chave_nfe = data.get('chaveNfe')
        self.segundo_codigo_barra = data.get('segundoCodigoBarra')
        self.indicador_reentrega = data.get('indicadorReentrega')

    def serialize(self):
        return {
            'id': self.id,
            'idMdfeMunicipioDescarrega': self.id_mdfe_municipio_descarrega,
            'chaveNfe': self.chave_nfe,
            'segundoCodigoBarra': self.segundo_codigo_barra,
            'indicadorReentrega': self.indicador_reentrega,
            'mdfeMunicipioDescarregaModel': self.mdfe_municipio_descarrega_model.serialize() if self.mdfe_municipio_descarrega_model else None,
        }

File: frontend/mdfe/lib/app/data/model/mdfe_informacao_nfe_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';

import 'package:mdfe/app/data/model/model_imports.dart';

class MdfeInformacaoNfeModel {
	int? id;
	int? idMdfeMunicipioDescarrega;
	String? chaveNfe;
	String? segundoCodigoBarra;
	int? indicadorReentrega;
	MdfeMunicipioDescarregaModel? mdfeMunicipioDescarregaModel;

	MdfeInformacaoNfeModel({
		this.id,
		this.idMdfeMunicipioDescarrega,
		this.chaveNfe,
		this.segundoCodigoBarra,
		this.indicadorReentrega,
		this.mdfeMunicipioDescarregaModel,
	});

	static List<String> dbColumns = <String>[
		'id',
		'chave_nfe',
		'segundo_codigo_barra',
		'indicador_reentrega',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Chave Nfe',
		'Segundo Codigo Barra',
		'Indicador Reentrega',
	];

	MdfeInformacaoNfeModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		idMdfeMunicipioDescarrega = jsonData['idMdfeMunicipioDescarrega'];
		chaveNfe = jsonData['chaveNfe'];
		segundoCodigoBarra = jsonData['segundoCodigoBarra'];
		indicadorReentrega = jsonData['indicadorReentrega'];
		mdfeMunicipioDescarregaModel = jsonData['mdfeMunicipioDescarregaModel'] == null ? MdfeMunicipioDescarregaModel() : MdfeMunicipioDescarregaModel.fromJson(jsonData['mdfeMunicipioDescarregaModel']);
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['idMdfeMunicipioDescarrega'] = idMdfeMunicipioDescarrega != 0 ? idMdfeMunicipioDescarrega : null;
		jsonData['chaveNfe'] = chaveNfe;
		jsonData['segundoCodigoBarra'] = segundoCodigoBarra;
		jsonData['indicadorReentrega'] = indicadorReentrega;
		jsonData['mdfeMunicipioDescarregaModel'] = mdfeMunicipioDescarregaModel?.toJson;
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		idMdfeMunicipioDescarrega = plutoRow.cells['idMdfeMunicipioDescarrega']?.value;
		chaveNfe = plutoRow.cells['chaveNfe']?.value;
		segundoCodigoBarra = plutoRow.cells['segundoCodigoBarra']?.value;
		indicadorReentrega = plutoRow.cells['indicadorReentrega']?.value;
		mdfeMunicipioDescarregaModel = MdfeMunicipioDescarregaModel();
		mdfeMunicipioDescarregaModel?.nomeMunicipio = plutoRow.cells['mdfeMunicipioDescarregaModel']?.value;
	}	

	MdfeInformacaoNfeModel clone() {
		return MdfeInformacaoNfeModel(
			id: id,
			idMdfeMunicipioDescarrega: idMdfeMunicipioDescarrega,
			chaveNfe: chaveNfe,
			segundoCodigoBarra: segundoCodigoBarra,
			indicadorReentrega: indicadorReentrega,
		);			
	}

	
}